src/libs/actions/Welcome.ts

```diff
@@ -98,47 +111,46 @@ Onyx.connect({
 
 /**
  * Shows a welcome action on first login
- *
- * @param {Object} params
- * @param {Object} params.routes
- * @param {Function} [params.showCreateMenu]
- * @param {Function} [params.showPopoverMenu]
  */
-function show({routes, showCreateMenu = () => {}, showPopoverMenu = () => {}}) {
+function show({routes, showCreateMenu = () => {}, showPopoverMenu = () => false}: ShowParams) {
     isReadyPromise.then(() => {
         if (!isFirstTimeNewExpensifyUser) {
             return;
         }
 
         // If we are rendering the SidebarScreen at the same time as a workspace route that means we've already created a workspace via workspace/new and should not open the global
         // create menu right now. We should also stay on the workspace page if that is our destination.
-        const topRoute = _.last(routes);
-        const isWorkspaceRoute = topRoute.name === 'Settings' && topRoute.params.path.includes('workspace');
-        const transitionRoute = _.find(routes, (route) => route.name === SCREENS.TRANSITION_BETWEEN_APPS);
-        const exitingToWorkspaceRoute = lodashGet(transitionRoute, 'params.exitTo', '') === 'workspace/new';
-        const openOnAdminRoom = lodashGet(topRoute, 'params.openOnAdminRoom', false);
-        const isDisplayingWorkspaceRoute = isWorkspaceRoute || exitingToWorkspaceRoute;
+        const topRoute = routes[routes.length - 1];
```

Comment:
I think we should check for out of bounds? _.last will handle gracefully, we need to do the same alternative here as well.